### PR TITLE
New Article: Sleigh odometer

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
It can probably benefit from 2 or 3 more sentences

preview:
state

### CODE REVIEW

File: 2022/incoming/sleigh-odometer.pod [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+my $counter = make_counter();
+
+$counter->(); $counter->(); $counter->(); $counter->();
+say $counter->();
+
+=end perl
+
+=head2 Final word from Santa
+
+These 2 previous code works well, butt there is definitely a simpler way!
```
Comment (bbrtj) [inline 2022/incoming/sleigh-odometer.pod:84]: Should probably be "pieces of code" or "code snippets"?

Also very unfortunate typo: "butt" ;)

File: 2022/incoming/sleigh-odometer.pod [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+say count();
+
+=end perl
+
+But sadly it does not work as expected because the distance is reset at each C<count()> call :/
+
+=head2 Asking experienced elves
+
+After asking some more experienced elf, the elf modified to implement a persistent variable via a "lexical trick":
```
Comment (bbrtj) [inline 2022/incoming/sleigh-odometer.pod:40]: "modified to implement" doesn't seem right. Maybe "decided to implement"?

**Comment by bbrtj** [comment]
I liked the article, good introduction to how state works. Left some comments, hope you don't mind. Some parts I wasn't so sure about, but since I'm not a native I decided not to bring them to your attention (they may be valid, but not constructions I'm familiar with).

File: 2022/incoming/sleigh-odometer.pod (hunk 1 above)
Comment (thibaultduponchelle) [reply 2022/incoming/sleigh-odometer.pod:84]: Good point thank you

File: 2022/incoming/sleigh-odometer.pod (hunk 2 above)
Comment (thibaultduponchelle) [reply 2022/incoming/sleigh-odometer.pod]: Fixed

**Comment by thibaultduponchelle** [comment]
I did the changes, if you're all OK with them, please approve @bbrtj

File: 2022/incoming/sleigh-odometer.pod [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+my $counter = make_counter();
+
+$counter->(); $counter->(); $counter->(); $counter->();
+say $counter->();
+
+=end perl
+
+=head2 Final word from Santa
+
+These 2 previous pieces of code works well, but there is definitely a simpler way!
```
Comment (bbrtj) [inline 2022/incoming/sleigh-odometer.pod:83]: For plural "pieces of code", I think it should be "work well", not "works well"

File: 2022/incoming/sleigh-odometer.pod [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+=head1 Sleigh odometer
+
+=head2 Customize the sleigh
+
+It's already December and elves are working hard to produce all toys for Christmas.
+
+On the other side, Santa is preparing his Sleigh, cleaning or customizing.
+ 
+This year, in addition to a temperature sensor, Santa asked to install an odometer to his sleigh so that he can count each delivery.
+For the record, the distance unit is not kilometers or miles nor inches but "elf little toe"... but it does not change what follows :)
```
Comment (bbrtj) [inline 2022/incoming/sleigh-odometer.pod:16]: Not sure, but maybe "not kilometers nor miles nor inches" would read better?

**Comment by bbrtj** [comment]
Additional two minor points, other than that looks good to me

File: 2022/incoming/sleigh-odometer.pod (hunk 3 above)
Comment (thibaultduponchelle) [reply 2022/incoming/sleigh-odometer.pod]: Fixed

File: 2022/incoming/sleigh-odometer.pod (hunk 4 above)
Comment (thibaultduponchelle) [reply 2022/incoming/sleigh-odometer.pod]: Fixed

**Comment by thibaultduponchelle** [comment]
Thank you @bbrtj, I fixed the points

**Review by bbrtj — APPROVED** [review]

File: 2022/incoming/sleigh-odometer.pod [hunk 5]
```diff
@@ -0,0 +1,105 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The Sleigh odometer
+Topic: state
+
+=encoding utf8
+
+=head1 Sleigh odometer
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:7]:
```suggestion
=head1 Sleigh Odometer
```

File: 2022/incoming/sleigh-odometer.pod [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+=head2 Customize the sleigh
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:9]:
```suggestion
=head2 Customize the Sleigh
```

File: 2022/incoming/sleigh-odometer.pod [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+It's already December and elves are working hard to produce all toys for Christmas.
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:11]:
```suggestion
It's already December and elves are working hard to produce all of the toys for Christmas.
```

File: 2022/incoming/sleigh-odometer.pod [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+On the other side, Santa is preparing his Sleigh, cleaning or customizing.
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:13]:
```suggestion
On the other side, Santa is preparing his sleigh, cleaning or customizing.
```

File: 2022/incoming/sleigh-odometer.pod [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+ 
+This year, in addition to a temperature sensor, Santa asked to install an odometer to his sleigh so that he can count each delivery.
+For the record, the distance unit is not kilometers nor miles nor inches but "elf little toe"... but it does not change what follows :)
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:16]:
```suggestion
For the record, the distance unit is neither kilometers nor miles nor inches but "elf little toe"... but it does not change what follows :)
```

File: 2022/incoming/sleigh-odometer.pod [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+In order to implement the counter of the odometer, an elf came with this code:
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:18]:
```suggestion
In order to implement the counter of the odometer, an elf came up with this code:
```

File: 2022/incoming/sleigh-odometer.pod [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+count(); count(); count(); count(); 
+say count();
+
+=end perl
+
+But sadly it does not work as expected because the distance is reset at each C<count()> call :/
+
+=head2 Asking experienced elves
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:37]:
```suggestion
=head2 Asking Experienced Elves
```

File: 2022/incoming/sleigh-odometer.pod [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+After asking some more experienced elf, the elf decided to implement a persistent variable via a "lexical trick":
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:39]:
```suggestion
After asking some more experienced elves, the elf decided to implement a persistent variable via a "lexical trick":
```

File: 2022/incoming/sleigh-odometer.pod [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+my $counter = make_counter();
+
+$counter->(); $counter->(); $counter->(); $counter->();
+say $counter->();
+
+=end perl
+
+=head2 Final word from Santa
```
Comment (oalders) [inline 2022/incoming/sleigh-odometer.pod:81]:
```suggestion
=head2 Final Word from Santa
```

**Review by oalders — APPROVED** [review]

**Pull request merged**